Examples/Run/Geant4/DD4hep/CMakeLists.txt

```diff
@@ -0,0 +1,17 @@
+add_executable(
+  ActsExampleDD4hepGeantinoRecording
```

Comment:
As a above. My suggestion is to name this executable `ActsSimGeantinoRecordingDD4hep`.

Reply:
This would be called `ActsExampleGeantinoRecordingDD4hep` in the follow-up PR.

Reply:
I would prefer to change the name only once and not again in the next PR. I am fine with `ActsExampleGeantinoRecordingDD4hep` as mentioned in another comment above.

Reply:
> Looks good. I have a few request, mostly naming related. I would also rename the helper header from `GeantinoRecordingBase.h` to `GeantinoRecordingImpl.h` since it does not provide base-class like functionality but the common implementation.

I like, probably we should rename this 'Base' to 'Impl' consistently, bur for this PR I will start here.